docs/apache-airflow/core-concepts/overview.rst

```diff
@@ -18,49 +18,151 @@
+* A *metadata database*, used by the *scheduler*, and *webserver* to store state of workflows and tasks.
+  Setting up a metadata database is described in :doc:`/howto/set-up-database` and is required for
+  Airflow to work.
 
+Optional components
+...................
 
-Basic airflow architecture
---------------------------
+There are also some optional components that are not present in the basic installation
 
-This is the basic architecture of Airflow that you'll see in simple installations:
+* Optional *worker*, which executes the tasks given to it by the scheduler. In the basic installation
+  worker might be part of the scheduler not a separate component. It can be run as a long running process
+  in the :doc:`CeleryExecutor <executor/celery>`, or as a POD in the
+  :doc:`KubernetesExecutor <executor/kubernetes>`.
+
+* Optional *triggerer*, which executes deferred tasks in an async-io event loop. In basic installation
+  where deferred tasks are not used, triggerer might not be present. More about deferring tasks can be
+  found in :doc:`/authoring-and-scheduling/deferring`.
+
+* Optional *dag processor*, which parses DAG files and synchronizes them into the
+  *metadata database* in basic installation *dag processor* might be part of the scheduler not
+  a separate component.
+
+* A folder of *DAG files*, is read by  *dag processor*, *workers* and *triggerer* when they are running.
+  If *dag processor* is present *scheduler** does not need to read the *DAG files* directly. More about
+  processing DAG files can be found in :doc:`/authoring-and-scheduling/dagfile-processing`
+
+* Optional folder of *plugins*. Plugins are a way to extend Airflow's functionality and by placing Python
+  files in the plugins folder you can extend Airflow functionality (similarly as via installed packages).
+  Plugins are read by the *scheduler*, *dag processor*, *triggerer* and *webserver*. More about
+  plugins can be found in :doc:`/authoring-and-scheduling/plugins`.
+
+Deploying Airflow components
+............................
+
+All the components are Python applications that can be deployed using various deployment mechanisms.
+
+They deployed Python applications can have extra *installed packages* installed in their Python environment.
+This is useful for example to install custom operators or sensors or extend Airflow functionality with custom
+plugins.
+
+While Airflow can be run in a single machine and with simple installation where only *scheduler* and
+*webserver* are deployed, Airflow is designed to be scalable and secure, and is able ot run in a distributed
+environment - where various components can run on different machines, with different security perimeters
+and can be scaled by running multiple instances of the components above. Also while single person can run and
+manage Airflow installation, Airflow Deployment in more complex setup can involve various roles of users
+as described in the :doc:`/security/security_model`.
```

Comment:
I split that into three separate paragraphs. - one about scalability one about security and following chapter about roles. I decided for now not to split out separate "roles" document. I just briefly mentioned the roles and described it here, and linked to the "security model" for more details. 

I also moved the following chapter 

```
Airflow itself is agnostic to what you're running - it will happily orchestrate and run anything,
either with high-level support from one of our providers, or directly as a command using the shell
or Python :doc:`operators`.
```

to the top-level description, as it feels that it should be there rather than in the "components" section.